extensions/sql-database-projects/src/dialogs/utils.ts

```diff
@@ -155,3 +155,32 @@ export function getDockerBaseImages(target: string): DockerImageInfo[] {
 		];
 	}
 }
+
+/**
+ * This adds the tag matching the target platform to make sure the correct image is used for the project's target platform when the docker base image is SQL Server.
+ * If the image is Edge, then no tag is appended
+ * @param projectTargetVersion target version of the project
+ * @param dockerImage selected base docker image without tag
+ * @param imageInfo docker image info of the selected docker image
+ * @returns dockerBaseImage with the appropriate image tag appended if there is one
+ */
+export function getDefaultDockerImageWithTag(projectTargetVersion: string, dockerImage: string, imageInfo?: DockerImageInfo,): string {
+	if (imageInfo?.displayName === constants.SqlServerDockerImageName) {
+		switch (projectTargetVersion) {
+			case constants.targetPlatformToVersion.get(SqlTargetPlatform.sqlServer2022):
```

Comment:
you don't need to add  -latest. by default -latest will be picked. if you don't add tag then the tag that matches with the host will be picked and that's the correct behavior we should keep

Reply:
If you meant get rid of the -latest in "2022-latest" so that the full image name is `mcr.microsoft.com/mssql/server:2022`, that throws an error of image not found.

If you meant get rid of the 2022 and only use the latest tag so that the image is `mcr.microsoft.com/mssql/server:latest`, then that container has Microsoft SQL Server 2017, which is why this change is adding the 2022-latest tag and 2019-latest tag for projects targeting those versions. 
![image](https://user-images.githubusercontent.com/31145923/184218710-b912986e-0cf2-4bf7-a5d9-fc919105b497.png)